scripts/github-workflow-e2e-matrix.ts

```diff
@@ -0,0 +1,36 @@
+import { spawnSync } from 'child_process';
+
+try {
+  const all = process.argv.includes('--all');
+  const affectedJson = runCommand(
+    'npx',
+    'nx',
+    'print-affected',
+    '--target=build-storybook',
+    all ? ' --all' : ''
+  );
+  const affected = JSON.parse(affectedJson);
+  const projects = affected.tasks
+    .map((task: any) => task.target.project)
+    .filter((project: string) => project && project !== 'storybook');
+  const include = projects.map((project: string) => {
+    return {
+      project,
+      token: `PERCY_TOKEN_${project.toUpperCase().replace(/-/g, '_')}_E2E`,
+    };
+  });
+  console.log(`::set-output name=projects::${JSON.stringify(projects)}`);
```

Comment:
The contents of these `console.log`s are critical to the functionality of your GitHub workflow, but to an untrained eye I doubt a future engineer would understand their importance. What if we moved the `::set-output` bits back into the workflow file (so that they communicate proper context to a future developer reading through the YAML file), and create a new `dev-infra` command that prints the affected projects in JSON format like this:

**e2e-pr.yml**
```
run: |
  echo ::set-output name=affectedStorybookProjects::$(npx skyux-dev print-affected --target=build-storybook)
```
The `skyux-dev print-affected` command would:
1. echo a JSON-parsable string of all the affected projects to the console in the following format: `{[projectName: string]: {screenshotAccessToken: string;}}` (we can add more fields in the future to accommodate more workflow tasks)
2. automatically exclude certain projects from the output (such as `ci` and `storybook`)

This same functionality could be applied to the existing workflow files so that all workflows use the same logic to determine affected projects.

**NOTE**
For the sake of this PR, we can keep the logic in the `/scripts/github-workflow-e2e-matrix.ts` file if you'd like, with the expectation that both scripts would be eventually moved to the `dev-infra` repo (so they can be properly unit tested). If you choose this route, the command in your YML file would look (something) like:
```
run: |
  echo ::set-output name=affectedStorybookProjects::$(npx ts-node scripts/github-workflow-e2e-matrix)
```

Reply:
The JSON is now generated in a `dev-infra` command, with one JSON document specifying the two workflow matrices as well as some simple variables.